tests/epyccel/recognised_functions/test_math_funcs.py

```diff
@@ -377,12 +382,735 @@ def floor_return_type_real(x):
         a = floor(x)
         return a
 
-    f1 = epyccel(floor_return_type_int)
+    f1 = epyccel(floor_return_type_int, language = language)
     x = randint(100)
     assert(isclose(f1(x) ,  floor_return_type_int(x), rtol=1e-15, atol=1e-15))
     assert(type(f1(x))  == type(floor_return_type_int(x))) # pylint: disable=unidiomatic-typecheck
 
-    f1 = epyccel(floor_return_type_real)
+    f1 = epyccel(floor_return_type_real, language = language)
     x = randint(100)
     assert(isclose(f1(x) ,  floor_return_type_real(x), rtol=1e-15, atol=1e-15))
     assert(type(f1(x))  == type(floor_return_type_real(x))) # pylint: disable=unidiomatic-typecheck
+
+#------------------------------- Ceil function -------------------------------#
+def test_ceil_call(language):
+    @types('real')
+    def ceil_call(x):
+        from math import ceil
+        return ceil(x)
+
+    f1 = epyccel(ceil_call, language = language)
+    x = rand()
+    assert(ceil_call(x) == f1(x))
+
+    assert isinstance(ceil_call(x), type(f1(x)))
+
+def test_ceil_phrase(language):
+    @types('real','real')
+    def ceil_phrase(x,y):
+        from math import ceil
+        a = ceil(x)*ceil(y)
+        return a
+
+    f2 = epyccel(ceil_phrase, language = language)
+    x = rand()
+    y = rand()
+    assert(isclose(ceil_phrase(x, y), f2(x, y), rtol=1e-15, atol=1e-15))
+
+#------------------------------- copysign function -------------------------------#
+
+def test_copysign_call(language = 'c'):
```

Comment:
Please mark this test as xfail for fortran. You can specify language specific xfails with the following syntax:
```
@pytest.mark.parametrize( 'language', (
        pytest.param("c", marks = pytest.mark.c),
        pytest.param("fortran", marks = [
            pytest.mark.xfail(reason="copysign not implemented"),
            pytest.mark.fortran]
        )
    )
)
def test_copysign_call(language):
```